core/src/main/java/com/graphhopper/util/PointList.java

```diff
@@ -216,6 +220,9 @@ public void add(double lat, double lon) {
     }
 
     public void add(double lat, double lon, double ele) {
+        if (this.isImmutable()) {
```

Comment:
To reduce redundancy I would move this into a ensureImmutability method

Reply:
Ah yes, makes sense, thanks :)

Reply:
Thanks, I name the method `ensureMutability` as we are making sure that the object is mutable, I think it is a bit clearer that way?